app/Listeners/Training/WaitingList/CheckWaitingListAccountMshipState.php

```diff
@@ -21,7 +21,9 @@ public function handle(AccountAltered $event)
         // ensure we have the latest data
         $account = $event->account->refresh();
 
-        $accountsWaitingList = $account->currentWaitingLists;
+        $accountsWaitingList = $account->currentWaitingLists->filter(function ($waitingList) {
+            return $waitingList->home_members_only === 1;
```

Comment:
This shouldn't need the === 1 because the 'home_members_only' itself is a boolean.